model-optimizer/extensions/middle/TensorIteratorInput.py

```diff
@@ -255,3 +255,183 @@ def replace_pattern(graph: Graph, match: dict):
+                ('LoopCond', 'LoopCondData'),
+                ('LoopCondData', 'SwitchInput', {'in': 1}),
+
+                ('EnterInput', 'EnterInputData'),
+                ('EnterInputData', 'MergeInput', {'in': 0}),
+                ('MergeInput', 'MergeInputData'),
+                ('MergeInputData', 'SwitchInput', {'in': 0}),
+                ('SwitchInput', 'SwitchInputData', {'out': 1}),
+                ('SwitchInputData', 'IdentityInput'),
+                ('IdentityInput', 'IdentityInputData'),
+                ('IdentityInputData', 'Gather', {'in': 0}),
+                ('IdentityInputData', 'NextIterationInput'),
+
+                ('InitIndex', 'InitIndexData'),
+                ('InitIndexData', 'EnterIndex'),
+                ('EnterIndex', 'EnterIndexData'),
+                ('EnterIndexData', 'MergeIndex', {'in': 0}),
+                ('MergeIndex', 'MergeIndexData'),
+                ('MergeIndexData', 'SwitchIndex', {'in': 0}),
+                ('MergeIndexData', 'Less', {'in': 0}),
+                ('LoopCondData', 'SwitchIndex', {'in': 1}),
+                ('SwitchIndex', 'SwitchIndexData', {'out': 1}),
+                ('SwitchIndexData', 'IdentityIndex'),
+                ('IdentityIndex', 'IdentityIndexData'),
+                ('IdentityIndexData', 'AddIndex', {'in': 0}),
+                ('AddIndex', 'AddIndexData'),
+                ('AddIndexData', 'NextIterationIndex'),
+                ('IndexDelta', 'IndexDeltaData'),
+                ('IndexDeltaData', 'AddIndex', {'in': 1}),
+
+                ('IdentityIndexData', 'UnsqueezeIndex'),
+                ('UnsqueezeIndex', 'UnsqueezeIndexData'),
+                ('UnsqueezeIndexData', 'Gather', {'in': 1}),
+                ('Axis', 'AxisData'),
+                ('AxisData', 'Gather', {'in': 2}),
+                ('Gather', 'GatherData'),
+                ('GatherData', 'SqueezeSlice'),
+                ('SqueezeSlice', 'SqueezeSliceData'),
+            ],
+        )
+
+    @staticmethod
+    def replace_pattern(graph: Graph, match: dict):
+        # retrieve attribute values for TensorIteratorInput node
+        init_time = match['InitIndex'].value.item(0)
+        time_step = match['IndexDelta'].value.item(0)
+        axis = match['Axis'].value.item(0)
+
+        # retrieve input and output nodes for TensorIteratorInput node
+        initial_input_node = match['EnterInput']
+        current_index_node = match['IdentityIndex']
+        size_node = match['EnterMaxIndex']
+        resulted_slice_node = match['SqueezeSlice']
+        resulted_slice_node_name = resulted_slice_node.soft_get('name', resulted_slice_node.id)
+
+        # create TensorIteratorInput node that reflects slicing of input for each time step along axis
+        ti_input_node = TensorIteratorInput(graph, dict(axis=axis, start=init_time, stride=time_step,
+                                                        name= resulted_slice_node_name + '/TensorIteratorInput_')
```

Comment:
nit: redundant space character and also, please, remove trailing "_" from the name

Reply:
Fixed.